### PR TITLE
MTS20: Added docker compose file for docker image initialization

### PR DESCRIPTION
**Pull request by sonam-yoezer**
## Description 
Added docker compose file for docker image initialization.

## File Changed

- [x] Created .env file.

- [x]  Created Dockerfile.

- [x] Created docker-compose.yml file.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+MYSQL_HOST=mysql
+MYSQL_PORT=3307
+# JWT Configuration
+JWT_SECRET=RG8geW91IGhhdmUgdG8gZGVhbCB3aXRoIEJhc2U2NCBmb3JtYXQ/IFRoZW4gdGhpcyBzaXRlIGlzIHBlcmZlY3QgZm9yIHlvdSEgVXNlIG91ciBzdXBlciBoYW5keSBvbmxpbmUgdG9vbCB0byBlbmNvZGUgb3IgZGVjb2RlIHlvdXIgZGF0YS4K
+JWT_ACCESS_TOKEN_EXPIRATION=300000
+JWT_REFRESH_TOKEN_EXPIRATION=604800000
+# CORS Configuration
+CORS_ORIGIN_URL=http://localhost:4200
+# Logging Configuration
+LOGGING_LEVEL_SQL=DEBUG
```
Comment (tanze123) [inline .env]: Add line break

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# Stage 2: Create lightweight runtime image
+FROM openjdk:21-slim
+# Set working directory
+WORKDIR /app
+# Copy the built jar from the builder stage
+COPY --from=builder /app/build/libs/*.jar /app/movie-ticketing-system-backend.jar
+# Expose the application port (adjust as needed)
+EXPOSE 8080
+# Run the jar file
+ENTRYPOINT ["java", "-jar", "/app/movie-ticketing-system-backend.jar"]
```
Comment (tanze123) [inline Dockerfile]: Add line break

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      - MYSQL_DATABASE=${MYSQL_DATABASE}
+      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
+      - MYSQL_ALLOW_EMPTY_PASSWORD=${MYSQL_ALLOW_EMPTY_PASSWORD}
+    # Persistent volume for database data
+    volumes:
+      - movie-ticketing-system-mysql-data:/var/lib/mysql
+# Persistent volume definitions
+# Ensures data persistence across container restarts
+volumes:
+  movie-ticketing-system-mysql-data:
```
Comment (tanze123) [inline docker-compose.yml]: Add line break

**Review by Tshering-Tee — APPROVED** [review]

File: .env [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+APP_NAME=movie-ticketing-system
+APP_PORT=8081
+# MySQL Configuration
+MYSQL_DATABASE=movie-ticketing-system
+MYSQL_ROOT_PASSWORD=1234
+MYSQL_ALLOW_EMPTY_PASSWORD=yes
+MYSQL_HOST=mysql
+MYSQL_PORT=3307
+# JWT Configuration
+JWT_SECRET=RG8geW91IGhhdmUgdG8gZGVhbCB3aXRoIEJhc2U2NCBmb3JtYXQ/IFRoZW4gdGhpcyBzaXRlIGlzIHBlcmZlY3QgZm9yIHlvdSEgVXNlIG91ciBzdXBlciBoYW5keSBvbmxpbmUgdG9vbCB0byBlbmNvZGUgb3IgZGVjb2RlIHlvdXIgZGF0YS4K
```
Comment (tanze123) [inline .env]: Remove JWT_SECRET

File: .env [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# Application Configuration
+APP_NAME=movie-ticketing-system
+APP_PORT=8081
+# MySQL Configuration
+MYSQL_DATABASE=movie-ticketing-system
+MYSQL_ROOT_PASSWORD=1234
```
Comment (tanze123) [inline .env]: Remove mysql password

**Review by tanze123 — APPROVED** [review]

**Pull request merged**